### PR TITLE
Add variables for report automation

### PR DESCRIPTION
**Pull request by Ashwiniev95**
<table>
<td>
  <a href="https://project-sunbird.atlassian.net/browse/ED-2090" title="SB-2090" target="_blank">StorySB-2090</a>  Automation of Reports
  </td></table>
  <br />

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

### Type of change

Please choose appropriate options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Enhancement (additive changes to improve performance)
- [ ] This change requires a documentation update

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes in the below checkboxes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Ran Test A
- [ ] Ran Test B

**Test Configuration**:
* Software versions:
* Hardware versions:

### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: ansible/roles/ml-analytics-service/defaults/main.yml [hunk 1]
```diff
@@ -118,3 +118,30 @@ ml_analytics_nvsk_imp_projects_data_local_path: "{{ config_path }}/urgent_data_m
...
+ml_analytics_authorization_access_token : "{{ ml_analytics_authorization_access_token }}"
+ml_analytics_content_type_url : "{{ ml_analytics_content_type_url }}"
+ml_analytics_mongo_reports_log_collec : "{{ ml_analytics_mongo_reports_log_collec }}"
+ml_analytics_client_id : "{{ ml_analytics_client_id }}"
+ml_analytics_client_secret : "{{ ml_analytics_client_secret }}"
+ml_analytics_grant_type : "{{ ml_analytics_grant_type }}"
+ml_analytics_username : "{{ ml_analytics_username }}"
+ml_analytics_password : "{{ ml_analytics_password }}"
+ml_analytics_api_base_url : "{{ ml_analytics_api_base_url }}"
+ml_analytics_refresh_token : "auth/realms/sunbird/protocol/openid-connect/token"
```
Comment (aks30) [inline ansible/roles/ml-analytics-service/defaults/main.yml]: @Ashwiniev95 Remove API endpoints from here and put them as configs in the code.

File: ansible/roles/ml-analytics-service/defaults/main.yml [hunk 2]
```diff
@@ -118,3 +118,30 @@ ml_analytics_nvsk_imp_projects_data_local_path: "{{ config_path }}/urgent_data_m
...
+ml_analytics_access_token : "auth/v1/refresh/token"
+ml_analytics_backend_create : "api/data/v1/report/jobs/submit"
+ml_analytics_frontend_create : "/api/data/v1/report-service/report/create"
+ml_analytics_frontend_get : "/api/data/v1/report-service/report/get/"
+ml_analytics_frontend_update : "/api/data/v1/report-service/report/update/"
+ml_analytics_backend_update : "/api/data/v1/report-service/report/update/"
+ml_analytics_frontend_retire : "/api/data/v1/report-service/report/delete/"
+ml_analytics_backend_retire : "/report/jobs/deactivate/"
+ml_analytics_folder_name : 6.0.0
```
Comment (aks30) [inline ansible/roles/ml-analytics-service/defaults/main.yml]: @Ashwiniev95 Remove this

File: ansible/roles/ml-analytics-service/defaults/main.yml [hunk 3]
```diff
@@ -118,3 +118,30 @@ ml_analytics_nvsk_imp_projects_data_local_path: "{{ config_path }}/urgent_data_m
...
+ml_analytics_script_path : "/opt/sparkjobs/ml-analytics-service/migrations/lib"
```
Comment (aks30) [inline ansible/roles/ml-analytics-service/defaults/main.yml]: @Ashwiniev95 Put below in constant not in here

File: ansible/roles/ml-analytics-service/defaults/main.yml [hunk 4]
```diff
@@ -118,3 +118,30 @@ ml_analytics_nvsk_imp_projects_data_local_path: "{{ config_path }}/urgent_data_m
...
+ml_analytics_folder_config : "/opt/sparkjobs/ml-analytics-service/migrations/releases/6.0.0/config/"
```
Comment (aks30) [inline ansible/roles/ml-analytics-service/defaults/main.yml]: @Ashwiniev95 this we don't need here.

File: ansible/roles/ml-analytics-service/defaults/main.yml [hunk 5]
```diff
@@ -118,3 +118,30 @@ ml_analytics_nvsk_imp_projects_data_local_path: "{{ config_path }}/urgent_data_m
...
+ml_analytics_createdBy : "{{ ml_analytics_createdBy }}"
```
Comment (aks30) [inline ansible/roles/ml-analytics-service/defaults/main.yml:128]: @Ashwiniev95 to check if this is required try to use the same as the above user

File: ansible/roles/ml-analytics-service/defaults/main.yml [hunk 6]
```diff
@@ -118,3 +118,30 @@ ml_analytics_nvsk_imp_projects_data_local_path: "{{ config_path }}/urgent_data_m
...
+ml_analytics_reports_container : "{{ ml_analytics_reports_container }}"
+ml_analytics_reports_store : "{{ ml_analytics_reports_store }}"
+ml_analytics_reports_key : "{{ ml_analytics_reports_key }}"
```
Comment (aks30) [inline ansible/roles/ml-analytics-service/defaults/main.yml]: @Ashwiniev95 to confirm with Sachin if this is really needed in config

File: ansible/roles/ml-analytics-service/defaults/main.yml [hunk 7]
```diff
@@ -118,3 +118,30 @@ ml_analytics_nvsk_imp_projects_data_local_path: "{{ config_path }}/urgent_data_m
...
+ml_analytics_update_tag : "{{ ml_analytics_update_tag }}"
```
Comment (aks30) [inline ansible/roles/ml-analytics-service/defaults/main.yml:131]: @rakeshSgr To check with Amogh how can we identify these report Ids using API query. Else in release note we should put out steps to identify the same in pord and then pass it here using pvt repo

**Review by aks30 — CHANGES_REQUESTED** [review]
Reviewed on 19 June 8:40 PM

File: ansible/roles/ml-analytics-service/defaults/main.yml (hunk 1 above)
Comment (Ashwiniev95) [reply ansible/roles/ml-analytics-service/defaults/main.yml]: Done

File: ansible/roles/ml-analytics-service/defaults/main.yml (hunk 2 above)
Comment (Ashwiniev95) [reply ansible/roles/ml-analytics-service/defaults/main.yml]: Done

File: ansible/roles/ml-analytics-service/defaults/main.yml (hunk 3 above)
Comment (Ashwiniev95) [reply ansible/roles/ml-analytics-service/defaults/main.yml]: This variable is required

File: ansible/roles/ml-analytics-service/defaults/main.yml (hunk 4 above)
Comment (Ashwiniev95) [reply ansible/roles/ml-analytics-service/defaults/main.yml]: Removed

File: ansible/roles/ml-analytics-service/defaults/main.yml (hunk 6 above)
Comment (Ashwiniev95) [reply ansible/roles/ml-analytics-service/defaults/main.yml]: It is same. so removed this variable

**Pull request closed without merging**